### PR TITLE
OC-338 split cucumber tests platform

### PR DESCRIPTION
**Pull request by MartijnSipsCGI**
The cucumber tests platform project is now split into 
- cucumber-tests-platform (original project)
- cucumber-tests-platform-common
- cucumber-tests-platform-micgorids
- cucumber-tests-platform-publiclighting

The cucumber-tests-platform-dlms project has been changed to cucumber-tests-platform-smartmetering

This pull request may only be merged when the github.com/SmartSocietyServices/Integration-Tests OC-338 branch and github.com/SmartSocietyServices/Smart-Meter-Integration OC-338 branch is merged!

### CODE REVIEW

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java [hunk 1]
```diff
@@ -7,9 +7,9 @@
  */
 package com.alliander.osgp.cucumber.platform.glue.steps.database.core;
```
Comment (bvdzwet) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java]: Wouldn't these steps belong in cucumber-tests-platform-smartmetering instead of in the general helper code?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java [hunk 2]
```diff
@@ -7,11 +7,11 @@
  */
 package com.alliander.osgp.cucumber.platform.glue.steps.database.core;
```
Comment (bvdzwet) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java]: Wouldn't these steps belong in cucumber-tests-platform-publiclighting instead of in the general helper code?

File: cucumber-tests-core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<!-- Copyright 2016 Smart Society Services B.V. Licensed under the Apache 
+  License, Version 2.0 (the "License"); you may not use this file except in 
+  compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0 -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.alliander</groupId>
```
Comment (MartijnSipsCGI) [inline cucumber-tests-core/pom.xml]: This is not in scope, and we want to merge this as soon as possible.

File: cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/Helpers.java [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+    /**
+     * When running automatic tests, it might be that not each project is
+     * started in tomcat. When the repo's are cleared at the beginning of a test
+     * run, you get some exceptions when the database wasn't found. Therefore
+     * this method is created to ignore that.
+     *
+     * @param repo
+     *            The repository to remove.
+     */
+    public static <T extends AbstractEntity> void cleanRepoAbstractEntity(final JpaRepository<T, Long> repo) {
```
Comment (MartijnSipsCGI) [inline cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/Helpers.java]: Removed

File: cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/Helpers.java [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+     * 'stamdata') are cleared, you get some exceptions when the database wasn't
+     * found. Therefore this method is created to ignore that
+     *
+     * @param repo
+     */
+    public static <T extends Serializable> void cleanRepoSerializable(final JpaRepository<T, Long> repo) {
+        try {
+            repo.deleteAllInBatch();
+        } catch (final Exception ex) {
+            LOGGER.error(ex.getMessage());
```
Comment (MartijnSipsCGI) [inline cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/Helpers.java]: This method is removed.

File: cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/ScenarioContext.java [hunk 6]
```diff
@@ -5,7 +5,7 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package com.alliander.osgp.cucumber.platform.core;
+package com.alliander.osgp.cucumber.core;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-core/src/test/java/com/alliander/osgp/cucumber/core/ScenarioContext.java]: Renamed to current()

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Defaults.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import com.alliander.osgp.adapter.ws.schema.core.configurationmanagement.LongTermIntervalType;
+import com.alliander.osgp.adapter.ws.schema.core.configurationmanagement.MeterType;
+import com.alliander.osgp.adapter.ws.schema.core.devicemanagement.DeviceFunctionGroup;
+
+public class Defaults extends com.alliander.osgp.cucumber.platform.Defaults {
+
+    public static final LightType CONFIGURATION_LIGHTTYPE = null;
+    public static final MeterType CONFIGURATION_METER_TYPE = null;
+    public static final LinkType CONFIGURATION_PREFERRED_LINKTYPE = null;
+    public static final LightType DEFAULT_CONFIGURATION_LIGHTTYPE = LightType.RELAY;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Defaults.java]: Not in scope, this was just moved from the Defaults.java from the cucumber-tests-platform project to the cucumber-tests-platform project.

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Defaults.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    public static final MeterType DEFAULT_CONFIGURATION_METER_TYPE = MeterType.AUX;
+    public static final LinkType DEFAULT_CONFIGURATION_PREFERRED_LINKTYPE = LinkType.ETHERNET;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Defaults.java]: Again, this was moved from the the Defaults.java from the cucumber-tests-platform project to the cucumber-tests-platform project. Adjusting this is out of scope.

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Keys.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package com.alliander.osgp.cucumber.platform.common;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Keys.java]: Added.

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Keys.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+
+public class Keys extends com.alliander.osgp.cucumber.platform.Keys {
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/Keys.java]: Updated

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/config/ws/CommonConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package com.alliander.osgp.cucumber.platform.common.config.ws;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/config/ws/CommonConfiguration.java]: Added license header and moved CommonConfiguration...

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/config/ws/CommonConfiguration.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.annotation.PropertySources;
+
+import com.alliander.osgp.cucumber.platform.config.AbstractPlatformApplicationConfiguration;
+
+@Configuration
+@PropertySources({ @PropertySource("classpath:cucumber-tests-platform-common.properties"),
+        @PropertySource(value = "file:/etc/osp/test/global-cucumber.properties", ignoreResourceNotFound = true),
+        @PropertySource(value = "file:/etc/osp/test/cucumber-tests-platform-common.properties", ignoreResourceNotFound = true), })
+public class CommonConfiguration extends AbstractPlatformApplicationConfiguration {
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/config/ws/CommonConfiguration.java]: Renamed

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceSteps.java [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.Map;
+
+import org.junit.Assert;
+
+import com.alliander.osgp.adapter.ws.schema.core.devicemanagement.Device;
+import com.alliander.osgp.cucumber.core.GlueBase;
+import com.alliander.osgp.cucumber.platform.Keys;
+
+public class DeviceSteps extends GlueBase {
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceSteps.java]: Wont change this, as this is old code.

File: cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/config/CoreMicrogridsPersistenceConfig.java [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+
+import com.alliander.osgp.cucumber.platform.config.ApplicationPersistenceConfiguration;
+import com.alliander.osgp.domain.core.repositories.DeviceRepository;
+import com.alliander.osgp.domain.microgrids.repositories.RtuDeviceRepository;
+
+@Configuration
+@EnableJpaRepositories(entityManagerFactoryRef = "entityMgrCoreMicrogrids", transactionManagerRef = "txMgrCoreMicrogrids", basePackageClasses = {
+        DeviceRepository.class, RtuDeviceRepository.class })
+public class CoreMicrogridsPersistenceConfig extends ApplicationPersistenceConfiguration {
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/config/CoreMicrogridsPersistenceConfig.java]: Out of scope.

File: cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/glue/hooks/ScenarioHooks.java [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    @Autowired
+    private Database databaseSteps;
+
+    /**
+     * Executed after each scenario.
+     *
+     * Order 1000 ensures this will be run as one of the first hooks after the
+     * scenario.
+     */
+    @After(order = 1000)
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/glue/hooks/ScenarioHooks.java]: It is just a value lower then 99999 to asure that the cucumber-tests-platform hook is executed as last.... Old code,

File: cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/glue/hooks/ScenarioHooks.java [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /**
+     * Executed before each scenario.
+     *
+     * Remove all stuff from the database before each test. Each test should
+     * stand on its own. Therefore you should guarantee that the scenario is
+     * complete.
+     *
+     * Order 0 ensures this will be run as first hook.
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/glue/hooks/ScenarioHooks.java]: Same as above,

File: cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/database/Database.java [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    public void prepareDatabaseForScenario() {
+        // Then remove stuff from osgp_adapter_protocol_iec61850
+        this.iec61850DeviceRepository.deleteAll();
+
+        // Then remove stuff from the osgp_adapter_ws_microgrids
+        this.rtuResponseDataRepository.deleteAll();
+
+        // Now remove all from the core.
+        this.taskRepository.deleteAll();
+        this.rtuDeviceRepository.deleteAll();
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/database/Database.java]: Yes, but because of the scenariohooks everything is removed, also the SSLD stuff..

File: cucumber-tests-platform-publiclighting/pom.xml [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+    <dependency>
+      <groupId>com.alliander</groupId>
+      <artifactId>cucumber-tests-platform</artifactId>
+      <type>test-jar</type>
+    </dependency>
+    <dependency>
+      <groupId>com.alliander</groupId>
+      <artifactId>cucumber-tests-platform-common</artifactId>
+      <type>test-jar</type>
+    </dependency>
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-publiclighting/pom.xml]: Removed, and the domain-publiclighting is not needed here

File: cucumber-tests-platform-publiclighting/pom.xml [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-adapter-ws-publiclighting</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-adapter-ws-tariffswitching</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-publiclighting/pom.xml]: No, we decided to take tariffswitching and publiclighting as publiclighting stream project. When it is required to split them up further we will do that at that time.

File: cucumber-tests-platform-publiclighting/src/test/java/com/alliander/osgp/cucumber/platform/publiclighting/glue/hooks/ScenarioHooks.java [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+    @Autowired
+    private Database databaseSteps;
+
+    /**
+     * Executed after each scenario.
+     *
+     * Order 99999 ensures this will be run as one of the first hooks after the
+     * scenario.
+     */
+    @After(order = 99999)
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-publiclighting/src/test/java/com/alliander/osgp/cucumber/platform/publiclighting/glue/hooks/ScenarioHooks.java]: This is just to order the scenariohooks. Old code by the way.

File: cucumber-tests-platform-publiclighting/src/test/java/com/alliander/osgp/cucumber/platform/publiclighting/glue/steps/database/adapterprotocoloslp/OslpDeviceSteps.java [hunk 21]
```diff
@@ -61,15 +60,17 @@ public void anSsldOslpDevice(final Map<String, String> settings) throws Throwabl
...
                     .findByDeviceIdentification(getString(expectedEntity, Keys.KEY_DEVICE_IDENTIFICATION));
 
             Assert.assertEquals(getString(expectedEntity, Keys.KEY_DEVICE_TYPE), entity.getDeviceType());
             Assert.assertEquals(getString(expectedEntity, Keys.KEY_DEVICE_UID), entity.getDeviceUid());
-            //Assert.assertEquals(getInteger(expectedEntity, Keys.RANDOM_DEVICE), entity.getRandomDevice());
-            //Assert.assertEquals(getInteger(expectedEntity, Keys.RANDOM_PLATFORM), entity.getRandomPlatform());
+            // Assert.assertEquals(getInteger(expectedEntity,
+            // Keys.RANDOM_DEVICE), entity.getRandomDevice());
+            // Assert.assertEquals(getInteger(expectedEntity,
+            // Keys.RANDOM_PLATFORM), entity.getRandomPlatform());
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-publiclighting/src/test/java/com/alliander/osgp/cucumber/platform/publiclighting/glue/steps/database/adapterprotocoloslp/OslpDeviceSteps.java]: Old code, but I removed it.

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/hooks/ScenarioHooks.java [hunk 22]
```diff
@@ -5,15 +5,15 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package com.alliander.osgp.cucumber.platform.dlms.glue.hooks;
+package com.alliander.osgp.cucumber.platform.smartmetering.glue.hooks;
```
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/hooks/ScenarioHooks.java]: Old code. But I updated it....

File: cucumber-tests-platform-common/pom.xml [hunk 23]
```diff
@@ -0,0 +1,150 @@
...
+      <groupId>com.alliander</groupId>
+      <artifactId>cucumber-tests-platform</artifactId>
+      <type>test-jar</type>
+    </dependency>
+    
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-adapter-ws-admin</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
```
Comment (bvdzwet) [inline cucumber-tests-platform-common/pom.xml]: Maven dependency analysis shows that this is an unused declared dependency.
It is probably cleaner to replace the dependency to osgp-adapter-ws-admin from the Platform repository by the used undeclared dependency to osgp-ws-admin from the Shared repository.

File: cucumber-tests-platform-common/pom.xml [hunk 24]
```diff
@@ -0,0 +1,150 @@
...
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-adapter-ws-core</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
```
Comment (bvdzwet) [inline cucumber-tests-platform-common/pom.xml]: Maven dependency analysis shows that this is an unused declared dependency.
It is probably cleaner to replace the dependency to osgp-adapter-ws-core from the Platform repository by the used undeclared dependency to osgp-ws-core from the Shared repository.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java (hunk 1 above)
Comment (MartijnSipsCGI) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java]: Well, I left it there because it is located in the core database. Didn't notice it by the way. But I moved it.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java (hunk 2 above)
Comment (MartijnSipsCGI) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java]: Moved this aswell.

File: cucumber-tests-platform/pom.xml [hunk 25]
```diff
@@ -103,84 +103,21 @@
...
-    </dependency>
-    <dependency>
-      <groupId>com.alliander.osgp</groupId>
-      <artifactId>osgp-adapter-ws-core</artifactId>
-      <classifier>classes</classifier>
-    </dependency>
-    <dependency>
       <groupId>com.alliander.osgp</groupId>
       <artifactId>osgp-adapter-ws-shared-db</artifactId>
     </dependency>
```
Comment (bvdzwet) [inline cucumber-tests-platform/pom.xml]: Maven dependency analysis shows that the osgp-adapter-ws-shared-db dependency declared here is not used.
To get a clearer picture of the actual dependencies I think it had better be removed,

File: cucumber-tests-platform-common/pom.xml (hunk 23 above)
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/pom.xml]: Updated.

File: cucumber-tests-platform-common/pom.xml (hunk 24 above)
Comment (MartijnSipsCGI) [inline cucumber-tests-platform-common/pom.xml]: Updated

File: cucumber-tests-platform-publiclighting/pom.xml [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-protocol-adapter-oslp</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
+    <dependency>
+      <groupId>com.alliander.osgp</groupId>
+      <artifactId>osgp-adapter-ws-publiclighting</artifactId>
+      <classifier>classes</classifier>
+    </dependency>
```
Comment (bvdzwet) [inline cucumber-tests-platform-publiclighting/pom.xml]: Maven dependency analysis shows the dependency to osgp-adapter-ws-publiclighting from the Platform repository is unused.
I think it is cleaner to replace this by the used undeclared dependency to osgp-ws-publiclighting from the Shared repository.

File: cucumber-tests-platform-publiclighting/pom.xml (hunk 19 above)
Comment (bvdzwet) [inline cucumber-tests-platform-publiclighting/pom.xml]: Maven dependency analysis shows the dependency to osgp-adapter-ws-tariffswitching from the Platform repository is unused.
I think it is cleaner to replace this by the used undeclared dependency to osgp-ws-tariffswitching from the Shared repository.

File: cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/database/Database.java (hunk 17 above)
Comment (bvdzwet) [inline cucumber-tests-platform-microgrids/src/test/java/com/alliander/osgp/cucumber/platform/microgrids/database/Database.java]: The scenario hooks are a bit of a mess. Methods like 'removeLeftOvers' and earlier attempts to just put the deletion of records in a Wait loop, hoping at some point the errors will stop, are strong indicators they could be restructured better.
As long as we still have flickering tests, failing because of database constraints sitting in the way of deleting data, these are of serious concern.
If not in this story, I think soon after it, there should be some serious cleanup, if indeed the possibly related test instabilities still exist.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java (hunk 1 above)
Comment (bvdzwet) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SmartMeterSteps.java]: The core database contains non-core stuff (e.g. specific device types) with strong relations to core-stuff (the generic devices in the database).
I think it is good that you moved these. Something to watch out for (after this restructuring) I think is that this may be one of the issues when deleting data in cucumber-tests-platform and the other dependent test projects (if not handled properly).

**Comment by MartijnSipsCGI** [comment]
retest this please

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/ConfigurationObjects.java [hunk 27]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (bvdzwet) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/ConfigurationObjects.java]: This class was removed from com.alliander.osgp.cucumber.platform.dlms.glue.steps.ws.smartmetering.smartmeteringadhoc on May 11, 2017 (https://github.com/OSGP/Integration-Tests/commit/4ba66765cc3cd316accb7c22a94768735fd5a57b), but it looks like this pull request will introduce it back into the code.
I think it should be removed here as well.

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/SpecificConfigurationObject.java [hunk 28]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (bvdzwet) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/SpecificConfigurationObject.java]: This class was removed from com.alliander.osgp.cucumber.platform.dlms.glue.steps.ws.smartmetering.smartmeteringadhoc on May 11, 2017 (https://github.com/OSGP/Integration-Tests/commit/4ba66765cc3cd316accb7c22a94768735fd5a57b), but it looks like this pull request will introduce it back into the code.
I think it should be removed here as well.
